### ISSUE TITLE
MS SQL call function (not stored procedure)

### ISSUE THREAD

**Issue by Demiden** [linq2db/linq2db#1857]
Hello.

I have a simple scalar-value sql function like 

**CREATE FUNCTION [dbo].[f_cost_1]
RETURNS int AS
BEGIN
return 2000
END**

And I`ve tried to call it from a .NETCoreApp 2.0 application like 

`var cost = db.ExecuteReader("f_cost_1"); //return empty reader`
or 
`var cost = db.ExecuteProc<int>("f_cost_1"); //return 0`
or
`var cost = db.Execute<int>("f_cost_1"); //return 0`

It works great with a **stored procedure** like 

**CREATE PROCEDURE [dbo].[f_cost_1]
as begin
select 2000
end**

So how can I call a **function** properly and get return int result?

**Comment by MaceWindu** [comment]
```cs
[Sql.Function("f_cost_1", ServerSideOnly = true)]
public static int CostFunction()
{
  throw new InvalidOperationException();
}

var cost = db.Select(() => CostFunction());
```

**Comment by Demiden** [comment]
Thanks for your answer but what if I can`t determine a function in advance?

A customers have an access to the database and may create and add any functions there. Our web application has an admin part where they can set the function names as property of some work object. And I can get only the name of that function (and expected result).

**Comment by sdanyliv** [comment]
```cs
var value = db.Execute<int>(“SELECT f_cost_1”);
```

**Comment by MaceWindu** [comment]
yep, Execute or FromSql or custom Sql.Extension. Just don't forget to apply `new SqlCommandBuilder().QuoteIdentifier()` to function name if you are taking it from user

**Comment by MaceWindu** [comment]
I will mark it as feature request. To review if it makes sense to instroduce some helper like `T Sql.Function<T>(name, params)`

**Comment by Demiden** [comment]
Thanks it`ll be great. Looking forward of this :)

**Comment by MaceWindu** [comment]
Then don't close it yet :)
